temporalcli/commandsgen/docs.go

```diff
@@ -75,43 +97,119 @@ func (w *docWriter) writeCommand(c *Command) {
 }
 
 func (w *docWriter) writeSubcommand(c *Command) {
-	fileName := strings.Split(c.FullName, " ")[1]
-	subCommand := strings.Join(strings.Split(c.FullName, " ")[2:], "")
-	w.fileMap[fileName].WriteString("## " + subCommand + "\n\n")
+	fileName := c.FileName
+	prefix := strings.Repeat("#", c.Depth)
+	w.fileMap[fileName].WriteString(prefix + " " + c.LeafName + "\n\n")
 	w.fileMap[fileName].WriteString(c.Description + "\n\n")
-	w.fileMap[fileName].WriteString("Use the following options to change the behavior of this command.\n\n")
 
-	// gather options from command and all options aviailable from parent commands
-	var allOptions = make([]Option, 0)
-	for _, options := range w.optionsStack {
-		allOptions = append(allOptions, options...)
-	}
+	if len(c.Children) == 0 {
```

Comment:
Great improvement here! The old code fails to handle a few edge cases with more complex subcommands.

How are `Parent` and `Children` used? I only see `Children` being used to check `if len(c.Children) == 0 {`, and I don't see the `Parent` information used anywhere. If that's the case, I feel like a simpler approach would be to keep a `IsLeafCommand` from `enrich.go` and use that for this condition here

Reply:
@yuandrew Thanks! `Parent` is used in `EnrichCommand()` for various things including setting `Children` on the `Parent`.

`Children` is used for the more [hierarchical docs gen style used in `tcld`](https://github.com/temporalio/tcld/pull/398/files#diff-119cb3ae9908bc291809b31d6ea1a292670f3c864de416a6ff822102eb847e9aR84) which we probably want to move to for `temporal` at some point.